code/z_adventurezones/space_casino.dm

```diff
@@ -0,0 +1,144 @@
+/*
+    Lythine's space casino prefab
+	Contents:
+	 Item slot machine
+	 Misc props
+*/
+
+// Item slot machine
+
+/obj/submachine/slot_machine/item
+	name = "Item Slot Machine"
+	desc = "A slot machine that produces items rather than money. Somehow."
+	icon_state = "slotsitem-off"
+	//mats = 50
+	var/uses = 0
+	icon_base = "slotsitem"
+
+	var/list/junktier = list( // junk tier, 68% chance
+		"/obj/item/a_gift/easter",
+		"/obj/item/raw_material/rock",
+		"/obj/item/balloon_animal",
+		"/obj/item/cigpacket",
+		"/obj/item/clothing/shoes/moon",
+		"/obj/item/fish/carp",
+		"/obj/item/instrument/bagpipe",
+		"/obj/item/clothing/under/gimmick/yay",
+		"/obj/item/scrap",
+		"/obj/item/paper_bin",
+		"/obj/item/item_box/gold_star",
+		"/obj/item/storage/box/costume/hotdog",
+		"/obj/critter/roach",
+		"/obj/item/device/light/flashlight",
+		"/obj/item/kitchen/utensil/knife",
+		"/obj/item/staple_gun",
+		"/obj/item/old_grenade/banana/cheese_sandwich",
+		"/obj/item/old_grenade/banana/banana_corndog",
+		"/obj/item/rubberduck"
+	)
+
+	var/list/usefultier = list( // half decent tier, 30% chance
+		"/obj/item/clothing/gloves/yellow",
+		"/obj/item/bat",
+		"/obj/item/reagent_containers/food/snacks/donkpocket/warm",
+		"/obj/item/device/flash",
+		"/obj/item/clothing/glasses/sunglasses",
+		"/obj/vehicle/skateboard",
+		"/obj/item/storage/firstaid/regular",
+		"/obj/item/clothing/shoes/sandal",
+		"/obj/item/cigpacket/random",
+		"/obj/item/clothing/mask/gas",
+		"/obj/critter/domestic_bee",
+		"/obj/item/storage/firstaid/crit"
+	)
+
+	var/list/raretier = list( // rare tier, 2% chance
+		"/obj/item/hand_tele",
+		"/obj/item/clothing/suit/armor/vest",
+		"/obj/item/gimmickbomb/hotdog",
+		"/obj/item/card/id/captains_spare",
+		"/obj/item/storage/banana_grenade_pouch",
+		"/obj/critter/wendigo", // have fun!
+		"/obj/item/artifact/teleport_wand",
+		"/obj/item/card/id/dabbing_license"
+	)
+
+	money_roll()
+		var/roll = rand(1,500)
+		var/exclamation = ""
+		var/win_sound = "sound/machines/ping.ogg"
+		var/obj/item/P = null
+
+		if (prob(src.uses))
+			src.emag_act(null, null) // bye bye!
+			return
+
+		if (roll <= 10) // rare tier, 2% chance
+			P = text2path(pick(raretier))
+			win_sound = "sound/misc/airraid_loop_short.ogg"
+			exclamation = "JACKPOT! "
+			src.uses += 20
+		else if (roll > 10 && roll <= 160) // half decent tier, 30% chance
+			P = text2path(pick(usefultier))
+			exclamation = "Big Winner! "
+		else // junk tier, 68% chance
+			P = text2path(pick(junktier))
+			exclamation = "Winner! "
+
+		if (P == null)
+			P = /obj/item/raw_material/rock
+		var/obj/item/prize = new P
```

Comment:
You are make an `/obj/` named P but using it to store text. Instead it should just be typeless since it's just holding our chosen type. We also dont need these conversions anymore
```suggestion
		var/prize_type = null

		if (prob(src.uses))
			src.emag_act(null, null) // bye bye!
			return

		if (roll <= 10) // rare tier, 2% chance
			prize_type = pick(raretier)
			win_sound = "sound/misc/airraid_loop_short.ogg"
			exclamation = "JACKPOT! "
			src.uses += 20
		else if (roll > 10 && roll <= 160) // half decent tier, 30% chance
			prize_type = pick(usefultier)
			exclamation = "Big Winner! "
		else // junk tier, 68% chance
			prize_type = pick(junktier)
			exclamation = "Winner! "

		if (!prize_type)
			prize_type = /obj/item/raw_material/rock
		var/obj/item/prize = new prize_type
```